From fbfc2ec3988a25dcd90a597de65c1ce26786d700 Mon Sep 17 00:00:00 2001
From: Alison Wang <alison.wang@nxp.com>
Date: Tue, 15 May 2018 14:58:37 +0800
Subject: [PATCH 264/706] drm: hdp: Add differential reference clock support

As the PHY supports single-ended reference clock or differential
reference clock, this patch uses differential reference clock instead
of single-ended reference clock.

Signed-off-by: Alison Wang <alison.wang@nxp.com>
(cherry picked from commit c3421c2e6bd5b53b1ade6a4dce0aeff13dfcebf6)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/gpu/drm/imx/hdp/API_AFE_mcu2_dp.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/gpu/drm/imx/hdp/API_AFE_mcu2_dp.c b/drivers/gpu/drm/imx/hdp/API_AFE_mcu2_dp.c
index 8f72b423935f..09d8191e53b8 100644
--- a/drivers/gpu/drm/imx/hdp/API_AFE_mcu2_dp.c
+++ b/drivers/gpu/drm/imx/hdp/API_AFE_mcu2_dp.c
@@ -499,10 +499,19 @@ void AFE_init(state_struct *state, int num_lanes,
 		       __func__, refclk);
 	}
 
+	val = Afe_read(state, PHY_PMA_CMN_CTRL1);
+	val = val & 0xFF8F;
+	Afe_write(state, PHY_PMA_CMN_CTRL1, val);
+	val = Afe_read(state, CMN_DIAG_ACYA);
+	Afe_write(state, CMN_DIAG_ACYA, 0x0100);
+
+	/* signal-ended reference clock */
+	/*
 	val = Afe_read(state, PHY_PMA_CMN_CTRL1);
 	val = val & 0xFF8F;
 	val = val | 0x0030;
 	Afe_write(state, PHY_PMA_CMN_CTRL1, val);
+	*/
 
 	if (state->edp != 0)
 		Afe_write(state, CMN_DIAG_CAL_CTRL, 0x0001);
-- 
2.17.1

